Docs: Fix broken link in the quickstart and fix error in code example, capitalize S in TypeScript (#30)
FuelLabs/fuel-docs: src/developer-quickstart.md

| script    | ❌                                   | ❌                    | ❌                           | ❌                        |
| library   | ✅ (via a contract or predicate) | ❌ | ❌                           | ✅                       |

> See [the chapter on program types](https://fuellabs.github.io/sway/v0.19.2/sway-program-types/index.html) for more information.

Reviewer: Should we use the latest version here? We sadly can't use `latest` in the link above but we can use `mater`:
```suggestion
See [the chapter on program types](https://fuellabs.github.io/sway/master/sway-program-types/index.html) for more information.
```

impl Counter for Contract {
    #[storage(read)]
    fn count() -> u64 {

>       storage.counter

Reviewer: Just 2 more spaces to get the right indentation :)
```suggestion
        storage.counter
```

>     }

Reviewer: Maybe a line break between the two methods:
```suggestion
    }

```

    #[storage(read, write)]

>     fn increment(){

Reviewer: Extra space after `()`
```suggestion
    fn increment() {
```

```sway
#[tokio::test]
async fn can_get_contract_id() {
    // Increment the counter
    let _result = instance.increment().call().await.unwrap();
    // Get the current value of the counter

>     let result = instance.counter().call().await.unwrap();

Reviewer: ```suggestion
    let result = instance.count().call().await.unwrap();
```

Reviewer: This one is totally on me - sorry.

```sway
#[tokio::test]
async fn can_get_contract_id() {

>     // Increment the counter

Reviewer: ```suggestion
    let (instance, _id) = get_contract_instance().await;
    // Increment the counter
```
